src/test/java/org/jenkinsci/plugins/workflow/DirectArtifactManagerFactory.java

```diff
@@ -58,10 +70,44 @@
  */
 public final class DirectArtifactManagerFactory extends ArtifactManagerFactory {
 
+    private static final Logger LOGGER = Logger.getLogger(DirectArtifactManagerFactory.class.getName());
     private static final AtomicInteger blockOpen = new AtomicInteger();
 
+    private final transient URL baseURL;
+
+    public DirectArtifactManagerFactory() throws Exception {
+        HttpServer server = ServerBootstrap.bootstrap().
```

Comment:
This is a pretty nifty change.   I dig it.

Reply:
FYI this was necessary because `RobustHTTPClient` works only on, well, HTTP(S) URLs. I have already clarified the core Javadoc to explain that external URLs must in fact be HTTP(S) (other protocols would hardly be sensible), so in order to _test_ handling of them, we need to be running a real HTTP server.